Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -1228,35 +1247,45 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
             
             try shouldContinueRegistration()
             
-            for (_, relationships) in combinedRelationships {
-                // Add parent <-> child edges to the URL tree
-                try relationships
-                    .compactMap(parentChildRelationship(from:))
-                    .sorted(by: Self.sortRelationshipsPreOrder)
-                    .forEach({ pair in
-                        // Add the relationship to the URL hierarchy
-                        let (parentRef, childRef) = pair
-                        // If the unique reference already exists, it's been added by another symbol graph
-                        // likely built for a different target platform, ignore it as it's the exact same symbol.
-                        if (try? symbolsURLHierarchy.parent(of: childRef)) == nil {
-                            do {
-                                try symbolsURLHierarchy.add(childRef, parent: parentRef)
-                            } catch let error as BidirectionalTree<ResolvedTopicReference>.Error {
-                                switch error {
-                                    // Some parents might not exist if they are types from other frameworks and
-                                    // those are not pulled into and available in the current symbol graph.
-                                case .nodeNotFound: break
-                                default: throw error
+            if LinkResolutionMigrationConfiguration.shouldSetUpHierarchyBasedLinkResolver {
```

Comment:
Instead of checking the conditional and force-unwrap `hierarchyBasedLinkResolver` every time, should we instead use an `if let hierarchyBasedLinkResolver = hierarchyBasedLinkResolver`?

Reply:
I updated various places to use `if let` to check if calls should be made to the `hierarchyBasedLinkResolver`. In some places it's still correct to check one of the `LinkResolutionMigrationConfiguration` static properties.